app/models/profile.rb

```diff
@@ -44,6 +44,7 @@ def decrypt_pii(password)
   def recover_pii(personal_key)
     encryptor = Encryption::Encryptors::PiiEncryptor.new(personal_key)
     decrypted_recovery_json = encryptor.decrypt(encrypted_pii_recovery, user_uuid: user.uuid)
+    return nil if JSON.parse(decrypted_recovery_json).nil?
```

Comment:
Could this have been implemented in `Pii::Attributes.new_from_json` to avoid double-parsing, and since we already have at least one check there already?

Here:

https://github.com/18F/identity-idp/blob/daf726599ffebfea3222273adec8a78e74d7748c/app/services/pii/attributes.rb#L22-L24

```rb
      return new if pii_json.blank?
      pii = JSON.parse(pii_json, symbolize_names: true)
      return new if pii.blank?
      new_from_hash(pii)
```

Reply:
It could, yeah

The competing factor is this is so far only known to affect recovery PII, and Pii::Attributes is used in more places and this is a quicker fix, so I opted towards the smaller scope, though it should definitely be given a fuller consideration in the follow-up work.